src/rooms/room_details/v1.rs

```diff
@@ -0,0 +1,95 @@
+//! [GET /_synapse/admin/v1/rooms/:room_id](https://github.com/element-hq/synapse/blob/master/docs/admin_api/rooms.md#room-details-api)
+use ruma::{
+    api::{metadata, request, response, Metadata},
+    events::room::{guest_access::GuestAccess, history_visibility::HistoryVisibility},
+    room::RoomType,
+    space::SpaceRoomJoinRule,
+    OwnedRoomAliasId, OwnedRoomId, OwnedUserId, UInt,
+};
+
+const METADATA: Metadata = metadata! {
+    method: GET,
+    rate_limited: false,
+    authentication: AccessToken,
+    history: {
+        unstable => "/_synapse/admin/v1/rooms/:room_id",
+    }
+};
+
+#[request]
+pub struct Request {
+    /// Alias or ID of the room to join.
+    #[ruma_api(path)]
+    pub room_id: OwnedRoomId,
+}
+
+#[response]
+pub struct Response {
+    /// Room ID
+    pub room_id: OwnedRoomId,
+
+    /// Room name
+    pub name: Option<String>,
+
+    /// Room topic
+    pub topic: Option<String>,
+
+    /// Room avatar
+    pub avatar: Option<String>,
```

Comment:
This should be an `Option<OwnedMxcUri>`.

Reply:
You're right :)